DemoApp/StreamChat/Components/DemoChatChannelListRouter.swift

```diff
@@ -518,6 +520,39 @@ final class DemoChatChannelListRouter: ChatChannelListRouter {
                         self?.showChannel(for: cid, at: message?.id)
                     }
                 }
+            }),
+            .init(title: "Test Multiple Chat Clients", handler: { [unowned self] _ in
+                // There should not be a secondary ChatClient, but even if there is,
+                // the current instance should not be affected by it.
+                let secondaryChatWrapper = StreamChatWrapper()
+                self.secondaryChatWrapper = secondaryChatWrapper
+                secondaryChatWrapper.logIn(as: .credentials(.hanSolo)) { error in
+                    if let error {
+                        log.error(error)
+                        return
+                    }
+                    guard let currentClient = StreamChatWrapper.shared.client else { return }
+                    guard let secondaryClient = secondaryChatWrapper.client else { return }
+                    // Trigger some DB changes
+                    Task {
+                        do {
+                            let currentChat = currentClient.makeChat(for: cid)
+                            let secondaryChat = secondaryClient.makeChat(for: cid)
+                            
+                            try await currentChat.get(watch: true)
+                            try await secondaryChat.get(watch: true)
+                            assert(currentChat.state.messages.count == secondaryChat.state.messages.count)
+                            
+                            await secondaryClient.disconnect()
+                            secondaryChatWrapper.connect(user: .credentials(.hanSolo), completion: { error in
+                                guard let error else { return }
+                                log.error(error)
+                            })
+                        } catch {
+                            print(error)
+                        }
+                    }
+                }
```

Comment:
Were we able to reproduce any issue with this test? If no, I would probably avoid adding this to the Demo App 🤔

Reply:
No luck with reproducing. I'll take it out.